app/controllers/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class UsersController < ApplicationController
+  skip_before_action :verify_authenticity_token
+  before_action :authenticate_user!
+
+  def index
+    @users = policy_scope(User)
+    authorize User
+    @roles = Role.all
+    @current_user = current_user
+  end
+
+  def update
+    @user = User.find(params[:id])
+    authorize @user
```

Comment:
why do you authorize the user before update it?

Reply:
The authorize method automatically infers that @user will have a matching PostPolicy class, and instantiates this class, handing in the current user and the given record.